tools/discord/bot.py

```diff
@@ -0,0 +1,246 @@
+    global NAND_PATH
+
+    fail_path = os.path.join(NAND_PATH, "fail")
+    if os.path.exists(fail_path):
+        with open(fail_path, "r") as f:
+            return f.read()
+    else:
+        return None
+
+
+async def dolphin_krkg() -> Optional[bytes]:
+    global NAND_PATH
+
+    krkg_path = os.path.join(NAND_PATH, "test.krkg")
+    if os.path.exists(krkg_path):
+        with open(krkg_path, "rb") as f:
+            return f.read()
+    else:
+        return None
+
+
+async def dolphin_generate_krkg(ghost: bytes, interaction: discord.Interaction):
+    global NAND_PATH
+
+    await dolphin_clear_io()
+    await dolphin_set_ghost(ghost)
+
+    await dolphin_run()
+
+    if await dolphin_ok():
+        krkg = await dolphin_krkg()
+        if not krkg:
+            await respond_bug_error(
+                interaction, "Dolphin returned OK, but there's no KRKG"
+            )
+            return
+
+        await respond_krkg_success(
+            interaction,
+            discord.File(os.path.join(NAND_PATH, "test.krkg"), filename="test.krkg"),
+        )
+        return
+
+    fail = await dolphin_fail()
+    if fail:
+        await respond_fail_error(interaction, fail)
+        return
+    # Empty string is falsey, leading to two situations where fail is False
+    elif os.path.exists(os.path.join(NAND_PATH, "fail")):
+        await respond_bug_error(
+            interaction, "Dolphin returned fail, but there's no explanation"
+        )
+        return
+
+    await respond_bug_error(interaction, "Dolphin returned nothing")
+
+
+# ================================
+#     RESPONSES
+# ================================
+
+
+async def respond_generic_error(
+    interaction: discord.Interaction, error: str, tip: Optional[str] = None
+):
+    embed = discord.Embed(color=0xFF595E, title="Error!", description=f"### {error}")
+    embed.set_footer(text=tip)
+    if (
+        interaction.response.type
+        == discord.InteractionResponseType.deferred_channel_message
+    ):
+        await interaction.followup.send(embed=embed, ephemeral=True)
+    else:
+        await interaction.response.send_message(embed=embed, ephemeral=True)
+
+
+async def respond_fail_error(
+    interaction: discord.Interaction, error: str, tip: Optional[str] = None
+):
+    embed = discord.Embed(color=0xFF595E, title="Fail!", description=f"### {error}")
+    embed.set_footer(text=tip)
+
+    if (
+        interaction.response.type
+        == discord.InteractionResponseType.deferred_channel_message
+    ):
+        await interaction.followup.send(embed=embed, ephemeral=True)
+    else:
+        await interaction.response.send_message(embed=embed, ephemeral=True)
+
+
+async def respond_bug_error(interaction: discord.Interaction, error: str):
+    embed = discord.Embed(color=0xFFCA3A, title="BUG!", description=f"### {error}")
+    embed.set_footer(
+        text="This should never be visible. "
+        "Please report this to a Kinoko developer if you see this!"
```

Comment:
"Hey! I got this output message!" inevitably leads to the follow-up question, "Okay, what RKG did you give it?" We should modify this text to explicitly tell the end user that they should provide the rkg affected.

Reply:
We have the RKG affected.

Reply:
Oh duh